packages/store/addon/-private/system/model/shim-model-class.ts

```diff
@@ -1,5 +1,24 @@
 import CoreStore from '../core-store';
 import { RelationshipSchema, AttributeSchema } from '../../ts-interfaces/record-data-schemas';
+import { Dict } from '../../ts-interfaces/utils';
+
+const AvailableShims = new WeakMap<CoreStore, Dict<ShimModelClass>>();
+
+export function getShimClass(store: CoreStore, modelName: string): ShimModelClass {
+  let shims = AvailableShims.get(store);
+
+  if (shims === undefined) {
+    shims = Object.create(null) as Dict<ShimModelClass>;
+    AvailableShims.set(store, shims);
+  }
+
+  let shim = shims[modelName];
+  if (shim === undefined) {
+    shim = shims[modelName] = new ShimModelClass(store, modelName);
+  }
+
+  return shim;
+}
 
 // Mimics the static apis of DSModel
 export default class ShimModelClass {
```

Comment:
Given my suggestions above and specifically what (2) would entail, you could define it so that here you had `ShimModelClass implements ModelStatic` or whatever you called it. That'd guarantee that instead of just mimicking the static API of DSModel, you'd be *guaranteed* to uphold it.

Reply:
I'm going to table this as an improvement we should do, and one I promise to get to soon :P

Reply:
I'm going to table this as something we should do (and do soon). It'll be a key step of the continuing work to pull the Model specific code bits into the model package.